Canonical sample not working with WildFly Swarm microprofile fraction 2016.8.1

In the canonical `pom.xml` I changed the Swarm profile to use the microprofile fraction : 

        <profile>
            <id>swarm</id>
            <dependencies>
                <dependency>
                    <groupId>org.wildfly.swarm</groupId>
                    <artifactId>microprofile</artifactId>
                    <version>2016.8.1</version>
                </dependency>
            </dependencies>
            <build>
                <plugins>
                    <plugin>
                        <groupId>org.wildfly.swarm</groupId>
                        <artifactId>wildfly-swarm-plugin</artifactId>
                        <configuration>
                            <properties>
                                <swarm.http.port>8081</swarm.http.port>
                                <swarm.context.path>/msTopCDs</swarm.context.path>
                            </properties>
                        </configuration>
                    </plugin>
                </plugins>
            </build>
        </profile>


This gives me an executable jar. When I run it, I get the following : 

2016-09-04 16:54:48,237 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-2) MSC000001: Failed to start service jboss.deployment.unit."microprofile-sample-canonical.war".WeldStartService: org.jboss.msc.service.StartException in service jboss.deployment.unit."microprofile-sample-canonical.war".WeldStartService: Failed to start service
	at org.jboss.msc.service.ServiceControllerImpl$StartTask.run(ServiceControllerImpl.java:1904)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.jboss.weld.exceptions.DeploymentException: WELD-001409: Ambiguous dependencies for type Integer with qualifiers @ConfigurationValue
  at injection point [BackedAnnotatedField] @Inject @ConfigurationValue org.wildfly.swarm.undertow.runtime.UndertowSocketBindingsCustomizer.httpsPort
  at org.wildfly.swarm.undertow.runtime.UndertowSocketBindingsCustomizer.httpsPort(UndertowSocketBindingsCustomizer.java:0)
  Possible dependencies: 
  - Producer Method [Integer] with qualifiers [@ConfigurationValue @Any] declared as [[BackedAnnotatedMethod] @Produces @ConfigurationValue @Dependent org.wildfly.swarm.container.runtime.cdi.ConfigurationValueProducer.produceIntegerConfigValue(InjectionPoint)],
  - Producer Method [Integer] with qualifiers [@ConfigurationValue @Any] declared as [[UnbackedAnnotatedMethod] @Produces @ConfigurationValue @Dependent org.wildfly.swarm.container.runtime.cdi.ConfigurationValueProducer.produceIntegerConfigValue(InjectionPoint)]

	at org.jboss.weld.bootstrap.Validator.validateInjectionPointForDeploymentProblems(Validator.java:367)
	at org.jboss.weld.bootstrap.Validator.validateInjectionPoint(Validator.java:281)
	at org.jboss.weld.bootstrap.Validator.validateGeneralBean(Validator.java:134)
	at org.jboss.weld.bootstrap.Validator.validateRIBean(Validator.java:155)
	at org.jboss.weld.bootstrap.Validator.validateBean(Validator.java:518)
	at org.jboss.weld.bootstrap.ConcurrentValidator$1.doWork(ConcurrentValidator.java:68)
	at org.jboss.weld.bootstrap.ConcurrentValidator$1.doWork(ConcurrentValidator.java:66)
	at org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:63)
	at org.jboss.weld.executor.IterativeWorkerTaskFactory$1.call(IterativeWorkerTaskFactory.java:56)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at org.jboss.threads.JBossThread.run(JBossThread.java:320)

--- a/README.md
+++ b/README.md
@@ -18,6 +18,7 @@ To build these examples you can just :
 
 These examples use Arquillian Tests. By default, Arquillian uses WildFly embedded, but other forms of containers can also be plugged in.
  
+* `mvn clean test -Pwildfly-swarm`  tests against a WildFly Swarm
 * `mvn clean test -Parquillian-wildfly-managed`  tests against an embedded WildFly
 * `mvn clean test -Parquillian-wildfly-remote` tests against a running WildFly
 * `mvn clean test -Parquillian-tomee-managed`  tests against an embedded Apache TomEE
@@ -51,6 +52,6 @@ These examples use Arquillian Tests. By default, Arquillian uses WildFly embedde
    
 Package the samples with WildFly Swarm using the following Maven command :    
 
-* `mvn clean package -Pswarm` 
+* `mvn clean package -Pwildfly-swarm` 
 
 This will create an executable Jar under the `target` directory. Just execute it with `java -jar target/microprofile-sample-canonical-swarm.jar` and invoke the microservice at http://localhost:8081/msTopCDs

--- a/microprofile-sample-canonical/pom.xml
+++ b/microprofile-sample-canonical/pom.xml
@@ -52,21 +52,7 @@
 
     <profiles>
         <profile>
-            <id>swarm</id>
-            <dependencies>
-                <dependency>
-                    <groupId>org.wildfly.swarm</groupId>
-                    <artifactId>undertow</artifactId>
-                </dependency>
-                <dependency>
-                    <groupId>org.wildfly.swarm</groupId>
-                    <artifactId>jaxrs-cdi</artifactId>
-                </dependency>
-                <dependency>
-                    <groupId>org.wildfly.swarm</groupId>
-                    <artifactId>jaxrs-jsonp</artifactId>
-                </dependency>
-            </dependencies>
+            <id>wildfly-swarm</id>
             <build>
                 <plugins>
                     <plugin>
@@ -79,9 +65,57 @@
                             </properties>
                         </configuration>
                     </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <version>${version.surefire.plugin}</version>
+                        <configuration>
+                            <skipTests>false</skipTests>
+                            <systemPropertyVariables>
+                                <arquillian.launch>arquillian-wildfly-swarm</arquillian.launch>
+                            </systemPropertyVariables>
+                        </configuration>
+                    </plugin>
                 </plugins>
             </build>
+            <!-- Temporary until 2016.10 released -->
+            <repositories>
+                <repository>
+                    <id>projectodd-snapshots</id>
+                    <name>Project:odd Snapshots from CI</name>
+                    <url>https://repository-projectodd.forge.cloudbees.com/snapshot</url>
+                    <releases>
+                        <enabled>false</enabled>
+                    </releases>
+                    <snapshots>
+                        <enabled>true</enabled>
+                    </snapshots>
+                </repository>
+            </repositories>
+            <pluginRepositories>
+                <pluginRepository>
+                    <id>projectodd-snapshots</id>
+                    <name>Project:odd Snapshots from CI</name>
+                    <url>https://repository-projectodd.forge.cloudbees.com/snapshot</url>
+                    <releases>
+                        <enabled>false</enabled>
+                    </releases>
+                    <snapshots>
+                        <enabled>true</enabled>
+                    </snapshots>
+                </pluginRepository>
+            </pluginRepositories>
+            <dependencies>
+                <dependency>
+                    <groupId>org.wildfly.swarm</groupId>
+                    <artifactId>microprofile</artifactId>
+                </dependency>
+                <dependency>
+                    <groupId>org.wildfly.swarm</groupId>
+                    <artifactId>arquillian</artifactId>
+                    <scope>test</scope>
+                </dependency>
+            </dependencies>
         </profile>
-
     </profiles>
 </project>

--- a/microprofile-sample-canonical/src/test/resources/arquillian.xml
+++ b/microprofile-sample-canonical/src/test/resources/arquillian.xml
@@ -38,6 +38,7 @@
 <arquillian xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://jboss.org/schema/arquillian"
             xsi:schemaLocation="http://jboss.org/schema/arquillian http://jboss.org/schema/arquillian/arquillian_1_0.xsd">
 
+    <container qualifier="arquillian-wildfly-swarm"/>
     <container qualifier="arquillian-wildfly-remote"/>
     <container qualifier="arquillian-wildfly-managed">
         <configuration>

--- a/microprofile-sample-swagger/pom.xml
+++ b/microprofile-sample-swagger/pom.xml
@@ -57,21 +57,7 @@
 
     <profiles>
         <profile>
-            <id>swarm</id>
-            <dependencies>
-                <dependency>
-                    <groupId>org.wildfly.swarm</groupId>
-                    <artifactId>undertow</artifactId>
-                </dependency>
-                <dependency>
-                    <groupId>org.wildfly.swarm</groupId>
-                    <artifactId>jaxrs-cdi</artifactId>
-                </dependency>
-                <dependency>
-                    <groupId>org.wildfly.swarm</groupId>
-                    <artifactId>jaxrs-jsonp</artifactId>
-                </dependency>
-            </dependencies>
+            <id>wildfly-swarm</id>
             <build>
                 <plugins>
                     <plugin>
@@ -84,9 +70,57 @@
                             </properties>
                         </configuration>
                     </plugin>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <version>${version.surefire.plugin}</version>
+                        <configuration>
+                            <skipTests>false</skipTests>
+                            <systemPropertyVariables>
+                                <arquillian.launch>arquillian-wildfly-swarm</arquillian.launch>
+                            </systemPropertyVariables>
+                        </configuration>
+                    </plugin>
                 </plugins>
             </build>
+            <!-- Temporary until 2016.10 released -->
+            <repositories>
+                <repository>
+                    <id>projectodd-snapshots</id>
+                    <name>Project:odd Snapshots from CI</name>
+                    <url>https://repository-projectodd.forge.cloudbees.com/snapshot</url>
+                    <releases>
+                        <enabled>false</enabled>
+                    </releases>
+                    <snapshots>
+                        <enabled>true</enabled>
+                    </snapshots>
+                </repository>
+            </repositories>
+            <pluginRepositories>
+                <pluginRepository>
+                    <id>projectodd-snapshots</id>
+                    <name>Project:odd Snapshots from CI</name>
+                    <url>https://repository-projectodd.forge.cloudbees.com/snapshot</url>
+                    <releases>
+                        <enabled>false</enabled>
+                    </releases>
+                    <snapshots>
+                        <enabled>true</enabled>
+                    </snapshots>
+                </pluginRepository>
+            </pluginRepositories>
+            <dependencies>
+                <dependency>
+                    <groupId>org.wildfly.swarm</groupId>
+                    <artifactId>microprofile</artifactId>
+                </dependency>
+                <dependency>
+                    <groupId>org.wildfly.swarm</groupId>
+                    <artifactId>arquillian</artifactId>
+                    <scope>test</scope>
+                </dependency>
+            </dependencies>
         </profile>
-
     </profiles>
 </project>

--- a/microprofile-sample-swagger/src/test/resources/arquillian.xml
+++ b/microprofile-sample-swagger/src/test/resources/arquillian.xml
@@ -38,6 +38,7 @@
 <arquillian xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://jboss.org/schema/arquillian"
             xsi:schemaLocation="http://jboss.org/schema/arquillian http://jboss.org/schema/arquillian/arquillian_1_0.xsd">
 
+    <container qualifier="arquillian-wildfly-swarm"/>
     <container qualifier="arquillian-wildfly-remote"/>
     <container qualifier="arquillian-wildfly-managed">
         <configuration>

--- a/pom.xml
+++ b/pom.xml
@@ -26,7 +26,7 @@
         <version.resteasy>3.0.18.Final</version.resteasy>
         <version.swagger>1.5.9</version.swagger>
         <!--Swarm-->
-        <version.wildfly-swarm>1.0.0.Final</version.wildfly-swarm>
+        <version.wildfly-swarm>2016.10-SNAPSHOT</version.wildfly-swarm>
         <!-- Test -->
         <version.arquillian>1.0.0.Alpha3</version.arquillian>
         <version.arquillian.wildfly.remote>8.2.1.Final</version.arquillian.wildfly.remote>
@@ -67,7 +67,7 @@
             <!-- WILDFLY SWARM -->
             <dependency>
                 <groupId>org.wildfly.swarm</groupId>
-                <artifactId>bom</artifactId>
+                <artifactId>bom-all</artifactId>
                 <version>${version.wildfly-swarm}</version>
                 <type>pom</type>
                 <scope>import</scope>

